app/main.py

```diff
@@ -1,3 +1,15 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    cache_dict = {}
+
+    def wrapper(*args) -> Any:
+        if args not in cache_dict:
+            print("Calculating new result")
+            result = func(*args)
+            cache_dict[args] = result
```

Comment:
BTW, you can do as follow
```suggestion
            cache_dict[args] = func(*args)
```